src/msg/simple/Accepter.cc

```diff
@@ -284,6 +291,49 @@ void Accepter::stop()
   done = false;
 }
 
+int Accepter::is_linked()
+{
+#if defined(ETHTOOL_GLINK) && defined(SIOCETHTOOL)
+  entity_addr_t iaddr = msgr->get_myaddr();
 
-
-
+  struct ifaddrs *ifList;
+  if(getifaddrs(&ifList) < 0){
+    assert(0 == "unable to fetch interfaces and addresses");
+  }
+  
+  string ifname;
+  for (struct ifaddrs *ifa = ifList; ifa != NULL; ifa = ifa->ifa_next) {
+    entity_addr_t ifaddr;
+    ifaddr.set_sockaddr(ifa->ifa_addr);
+    if (iaddr.is_same_host(ifaddr)) {
+      ifname = std::string(ifa->ifa_name);
```

Comment:
I would propose to skip creation of temporary std::string.
     ifname = ifa->ifa_name;